### DISCUSSION TITLE
How does LRR determine if some metadata is a timestamp?

### DISCUSSION THREAD

**Discussion by chierinyan** [Difegue/LANraragi#981] [Q&A] [answered]
I noticed that `timestamp` and `date_added` are treated as timestamps. However, when I tried to add another timestamp with namespace `date_uploaded`, it was handled as is.
Are these behaviors hardcoded, or can we somehow tell LRR to trade some namespace as timestamps or other datatypes?
![screenshot20240606042144](https://github.com/Difegue/LANraragi/assets/71999828/46268891-ebf1-445f-9da3-bc920733ee76)
![screenshot20240606042126](https://github.com/Difegue/LANraragi/assets/71999828/2cf74a91-3db6-43a2-a2b5-e7f52162f705)

**Comment by Difegue** [comment, marked as answer]
It's hardcoded, although the conversion is really just done clientside: https://github.com/Difegue/LANraragi/blob/5e1df8f94e9e729d9da50f7770d1b7c6d22c9cca/public/js/common.js#L126

**Reply by chierinyan** [reply]
Thanks for the explanation. Maybe we can hardcode some kind of pattern instead,  e.g. `time*` and  `date*`?

**Reply by Difegue** [reply]
Sure, I'd merge a PR with that
